### PR TITLE
Stats-Interactives CSS Transitions

### PR DESCRIPTION
**Pull request by nbuonin**

### CODE REVIEW

File: content/articles/stats-css-transitions.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+poster_source: ""
+bookendanimal: ""
+---
+For the Stats Interactives project, I was given a challenge that initially I
+wasn’t sure how to solve.  
+
+First, some background: Stats Interactives is a set of four simulations used in
+Quantitative Analysis I at the School for International and Public Affairs.
+The course teaches statistics to future policy makers, so that they’ll have the
+skills to understand research that they would encounter in their careers.
```
Comment (sdreher) [inline content/articles/stats-css-transitions.md]: "to understand research they will encounter"

File: content/articles/stats-css-transitions.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+For one interactive, our faculty partner Doru Cojoc invented a particularly
```
Comment (sdreher) [inline content/articles/stats-css-transitions.md]: Dr. Doru Cojoc

File: content/articles/stats-css-transitions.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+insightful method for teaching students linear regression.  He found that his
+students would struggle to understand the distribution of values around the
+predicted mean in the classic X-Y representation of graphs.  By rotating the
+graphs, such that the Y axes would align vertically, it would make it easier
```
Comment (sdreher) [inline content/articles/stats-css-transitions.md]: suggested: ", students could more easily see the relationship"

File: content/articles/stats-css-transitions.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+for students to see the relationship.  In class, he would demonstrate this
+relationship statically - first showing students the graphs with the Y axes
+aligned, and then showing them in the classical view with the X axes
+aligned.
+
+The challenge given to our team was how could we help reinforce this
```
Comment (sdreher) [inline content/articles/stats-css-transitions.md]: suggested: "was reinforcing this juxtaposition in the online environment"

File: content/articles/stats-css-transitions.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+juxtaposition for graphs.  Initially we thought we would show one, than the
+other - just as Prof. Cojoc would in class - one slide, than another.  Prof.
```
Comment (sdreher) [inline content/articles/stats-css-transitions.md]: Dr. or Prof? Both?

File: content/articles/stats-css-transitions.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+Cojoc wanted more though. He imagined these as an animation, first showing the
+graphs vertically, then rotating them horizontally.
+
+As the developer on the project, I wasn’t sure that there was a way to do what
+he imagined. Thinking purely in terms of animation, I was thinking that I’d
+need to paint all the page content onto a canvas element, so that they could be
+animated.  On its face though, this wasn’t feasible.  It would make the content
+inaccessible, and there wasn’t time to add such a layer of complexity to the
+app,  I knew that the CSS spec included animations and transitions, but I
```
Comment (sdreher) [inline content/articles/stats-css-transitions.md]: end sentence here: `app.`

File: content/articles/stats-css-transitions.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+thought these would not apply because the graphs in the app were rendered as
+SVG’s.  Once I took a closer look at the
```
Comment (sdreher) [inline content/articles/stats-css-transitions.md]: SVGs.

File: content/articles/stats-css-transitions.md [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+[docs](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Transitions/Using_CSS_transitions),
+I realized that I could make use of these parts of the CSS API, and achieve
+what Doru had in mind.
```
Comment (sdreher) [inline content/articles/stats-css-transitions.md]: Dr./Prof?

File: content/articles/stats-css-transitions.md [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+The initial layout would have the css transition applied to it while the
+subsequent layout was laying in wait, its opacity set to 0.  Once the parts of
+the first layout were moved into position they would fade out, and the new
+layout would fade in.
+
+Within a single transition, there are a few things that need to happen.
+Perhaps one element needs to rotate, perhaps another needs to rotate and
+reflect. The CSS transition syntax makes this easy to handle with `transform`
+and `transition` keywords.  I thought of `transform` as the "what" - what
+should this element do; and ‘transition’ as the "how" - specifically which
```
Comment (sdreher) [inline content/articles/stats-css-transitions.md]: backticks around `transition` here.

File: content/articles/stats-css-transitions.md [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+`transform`’s should happen when.  It’s also worth mentioning how using nested
+selectors in SCSS made writing these transitions a bit more easier.  The entire
```
Comment (sdreher) [inline content/articles/stats-css-transitions.md]: "a bit easier", drop the more

**Comment by nbuonin** [comment]
Thanks @sdreher I took all your edits.

**Pull request merged**